На ревью final

AlinaLyubeznova opened the pull request.

AlinaLyubeznova committed:
workflows

AlinaLyubeznova committed:
Добавлены новые стили, разметка, контент

AlinaLyubeznova committed:
Добавлены новые стили, разметка, контент

AlinaLyubeznova committed:
updated tests

AlinaLyubeznova committed:
Добавлены новые стили, разметка, контент

Savallera commented on index.html:
```diff
@@ -1,18 +1,24 @@
+    <link rel="stylesheet" href="./styles/animations.css">
+    <link rel="stylesheet" href="./styles/themes.css">
+    <link rel="icon" href="./images/favicon.ico" sizes="any">
+    <link rel="icon" href="./svg/favicon.svg" type="image/svg+xml">
```
**Отлично** фавикон меняет цвет в зависимости от темы

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+          </div>
+        <div class="card__buttons">
+          <button class="card__icon-button" type="button">
+            <svg class="like-icon" xmlns="http://www.w3.org/2000/svg" width="39" height="36" viewBox="0 0 39 36" fill="none">
```
**Надо исправить**
             Данный элемент без смысловой нагрузки, было бы хорошим решением добавить `aria-hidden="true"`, для того, чтобы скрыть его от скринридеров.
             Подробнее => https://doka.guide/a11y/aria-hidden/

             Необходимо устранить данное замечание для всех аналогичных элементов в коде ниже.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+          <img 
+          src="./images/image-two.jpg" 
+          alt="Лампа с мотивирующей надписью на английском языке рядом с ноутбуком"
+          class="card__image card__image-first"
```
**Надо исправить** Классы `card__image-first`, `card__image-second` и т.д. следует удалить. Название класса должно отражать и обосновывать его же использование. Скорректируйте именования для фильтров изображений по их стилизации, чтобы визуально было наглядно в коде имя класса.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+              fill="var(--heart-fill-color)"/>
+            </svg>
+          </button>
+          <button class="card__like-button" type="button">
```
**Надо исправить**

Кнопки `Like`, `Сохранить на память` и `Ок` одинаково оформлены в макете, добавьте им общий класс `button`, в котором опишите все состояния по макету `базовый :hover :focus, :focus-visible`. 

Необходимо устранить данное замечание для всех аналогичных элементов в коде ниже.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+        </div>
+      </section>
+    </main>
+    <footer>
```
**Надо исправить** Данный раздел не является подвалом страницы, `footer` необходимо убрать, после перенести контент в `main`

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+    </main>
+    <footer>
+      <button 
+        class="card__save-button" 
```
**Надо исправить**

Кнопки `Like`, `Сохранить на память` и `Ок` одинаково оформлены в макете, добавьте им общий класс `button`, в котором опишите все состояния по макету `базовый :hover :focus, :focus-visible`. Необходимо устранить данное замечание для всех аналогичных элементов в коде ниже.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+          <svg class="button__save-icon dialog_floppy" width="29" height="28" viewBox="0 0 29 28">
+            <use href="./svg/floppy.svg#floppy" />
+          </svg>
+          <span class="button__text dialog_text">Вставьте дискету, чтобы продолжить</span>
```
**Надо исправить** 
1. Для улучшения семантики текстовые элементы оформляют параграфом. `span` - элемент, не имеющий семантического значения.
2. Класс `button__text` здесь следует удалить, т.к. это элемент базовой кнопки.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+        type="button" 
+        onclick="window['dialog-id'].showModal();"
+        aria-controls="dialog-id">
+        <svg class="button__save-icon" width="29" height="28" viewBox="0 0 29 28">
```
**Можно лучше** Используйте базовый класс для иконки здесь `floppy-icon`, для того, чтобы описать базовые сво-ва, т.к. этот класс будет отдельным от кнопки и его можно будет использовать в диалоговом окне, а дополнительным классов скорректировать отличия.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+        <span class="button__text">Сохранить на память</span>
+      </button>
+      <dialog id="dialog-id">
+        <form method="dialog" class="dialog_form">
```
**Отлично** Используется правильный метод и форма для диалогового окна

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+            class="card__like-button dialog_button"
+            type="submit" 
+            onclick="window['dialog-id'].close()">
+            <span class="button__text dialog_text">OK</span>
```
**Надо исправить** 
           Для стилизации заглавными буквами лучше использовать специальное свойство [text-transform](https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform). 
           Чем это лучше? Бывает, что заглавные буквы в макете меняют на обычные, и тогда приходится вносить ручные правки, тогда как поправить свойство стиля куда быстрее.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+          <button
+            class="card__like-button dialog_button"
+            type="submit" 
+            onclick="window['dialog-id'].close()">
```
**Надо исправить**  Удалите атрибут `onclick` из элемента `button`, у кнопки верный тип, а у формы установлен метод, при щелчке будет обращение к форме и она закроется.

Savallera commented on index.html:
```diff
@@ -39,7 +45,422 @@
+          </svg>
+          <span class="button__text dialog_text">Вставьте дискету, чтобы продолжить</span>
+          <button
+            class="card__like-button dialog_button"
```
**Надо исправить**

Кнопки `Like`, `Сохранить на память` и `Ок` одинаково оформлены в макете, добавьте им общий класс `button`, в котором опишите все состояния по макету `базовый :hover :focus, :focus-visible`. Необходимо устранить данное замечание для всех аналогичных элементов в коде ниже.

Savallera commented on styles/variables.css:
```diff
@@ -1 +1,17 @@
+
+:root {
+  --main-font: 'Inter', sans-serif;
+  --accent-font: 'PressStart2P', fantasy;
```
**Отлично** 
  Используется альтернативный шрифт, это полезно, в случае, если браузер не подгрузит основной.

Savallera commented on styles/themes.css:
```diff
@@ -4,3 +4,47 @@
+  --card-text-font-weight: 465;
+  --card-title-font-weight: 785;
+  --heart-contour-color: #00cc14;
+  --heart-fill-color: #f00;
```
**Надо исправить** Данный цвет не меняется, он указан в файле variables.css, здесь следует удалить

Savallera commented on styles/themes.css:
```diff
@@ -4,3 +4,47 @@
+  --card-text-font-weight: 400;
+  --card-title-font-weight: 700;
+  --heart-contour-color: #000;
+  --heart-fill-color: #f00;
```
**Надо исправить** Данный цвет не меняется, он указан в файле variables.css, здесь следует удалить

Savallera commented on styles/themes.css:
```diff
@@ -4,3 +4,47 @@
+    --card-text-font-weight: 465;
+    --card-title-font-weight: 785;
+    --heart-contour-color: #00cc14;
+    --heart-fill-color: #f00;
```
**Надо исправить** Данный цвет не меняется, он указан в файле variables.css, здесь следует удалить

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+
+.like-icon .main-body {
+  fill: var(--bg-color);
+  transition: fill 0.3s 0.03s linear;
```
**Надо исправить** Удалите задержку для контура, здесь ее нет в задании

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+  transition: fill 0.3s 0.03s linear;
+}
+
+.like-icon:active .core {
```
**Надо исправить** Не добавлен параметр плавности перехода по заданию

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+  fill: var(--heart-fill-color);
+}
+
+.like-icon:active .main-body {
```
**Надо исправить** Не добавлен параметр плавности перехода по заданию

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+  transition: fill 0.3s 0.05s linear;
+}
+
+.like-icon.is-liked .contour {
```
**Надо исправить** Не добавлено описание селектора `.like-icon .contour {}`

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+
+.like-icon.is-liked .heart {
+  transform-origin: center;
+  animation: heart-scale 0.3s 0.1s ease-in reverse;
```
**Надо исправить** Параметры анимации не соотв. заданию (скорректируйте направление)

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+}
+
+.like-icon.is-liked .heart {
+  transform-origin: center;
```
**Надо исправить** Опишите данное состояние в дефолтном селекторе `.like-icon {}`, чтобы работало сразу

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,84 @@
+}
+
+.like-icon.is-liked .sparks {
+  animation: sparks 0.3s 0.3s ease-in reverse;
```
**Надо исправить** Параметры анимации не соотв. заданию (скорректируйте направление)

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  display: flex;
+  flex-direction: column;
+  align-items: center;
 }
```
**Надо исправить** 
Задайте для `.page` пропишите `background-attachment: fixed;`, чтобы фон оставался на месте при прокрутке страницы. https://www.notion.so/praktikum/067551a613d34b179bbad89696292c5e

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  margin-block: 102px 104px;
+  text-align: center;
+  inline-size: clamp(23.188rem, 30.516vw + 16.035rem, 43.5rem);
+  block-size: clamp(20.188rem, 3.85vw + 19.285rem, 22.75rem);
```
**Надо исправить**
 Для блоков с динамически изменяющимся контентом не следует создавать жесткое ограничение высоты. Если контента будет много он вывалится из блока, если мало, то блок будет почти пуст.

 * Если блок содержит изображение на фоне или абсолютно спозиционированные элементы, то тогда лучше используйте свойство `min-height`.  https://doka.guide/css/height/#kak-eto-ponyat
 * Если блок содержит текстовой контент не используется указание высоты совсем.
 * Вместо фиксированной высоты хорошо кнопкам добавлять внутренние отступы `padding`

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  text-align: center;
+  inline-size: clamp(23.188rem, 30.516vw + 16.035rem, 43.5rem);
+  block-size: clamp(20.188rem, 3.85vw + 19.285rem, 22.75rem);
+  border: 2px solid var(--accent-color);
```
**Можно лучше**
 В качестве цвета удобно использовать [`currentColor`](https://doka.guide/css/currentcolor/). Это хорошо тем, что при смене цвета текста в проекте, не нужно будет в каждом отдельном селекторе его также корректировать. Только один раз.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+.theme-menu__button:hover {
+  cursor: pointer;
+  opacity: 0.6;
+  transition: opacity 0.3s ease;
```
**Надо исправить** Перенесите параметры плавности в базовый селектор `.theme-menu__button`, чтбы работал сразу

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+
+.theme-menu__button:disabled {
+  opacity: 0.6;
+  transition: opacity 0.3s ease;
```
**Надо исправить** Дублирование после удалите

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+
+.theme-menu__button:disabled::before {
+  content: '▶';
+  opacity: 0.6;
```
**Надо исправить** Удалите лишнюю прозрачность - удвоилась, т.к. от кнопки наследуется

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  content: '▶';
+  opacity: 0.6;
+  animation: moving-arrow 0.5s ease-in-out infinite alternate;
+  position: absolute;
```
**Надо исправить** 
 При использовании абсолютного позиционирования необходимо указывать сразу две координаты. 
 Например, `inset-block-start` и `inset-inline-start`. 
 Если задавать координату только по одной оси - может возникнуть нестандартное отображение в некоторых браузерах, например в браузере Safari. Причем даже в последних его версиях.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  transition: transform 0.3s ease;
+}
+
+.theme-menu__button:focus-visible::after {
```
**Надо исправить** Удалите все дублирования от селектора `.theme-menu__button::after {}`. В данном сво-ве опишите только, что отличается.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  border: 2px solid var(--accent-color);
+  display: flex;
+  flex-direction: column;
+  overflow: hidden;
```
**Надо исправить** Параметры скрытия здесь не понадобяться, по заданию нет в карточке элементов выпадающих за пределы родителя.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  opacity: 0.5;
+}
+
+@supports (-webkit-text-stroke: 0.5px var(--label-text-stroke)) {
```
**Надо исправить** Скорректируйте параметр по макету - 1пиксель, также добавьте еще проверку на сво-во `text-stroke` без префикса, хорошо применить оператор `or`

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+}
+
+.card__image {
+  inline-size: 100%;
```
**Надо исправить** Добавьте также к изображению соотношение сторон, в случае смены изображение на любое другое, чтобы сохранялся квадрат. aspect-ratio https://doka.guide/css/aspect-ratio/

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+
+.card__image-seventh {
+  filter: hue-rotate(90deg) brightness(120%) saturate(150%);
+}
```
**Отлично** Используются красивые режимы наложения

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  filter: hue-rotate(90deg) brightness(120%) saturate(150%);
+}
+
+.card__text {
```
**Надо исправить** Параметры текста не соотв. макету.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  position: relative;
+  z-index: 1;
+  color: var(--button-text);
+  mix-blend-mode: difference;
```
**Отлично** Используется правильный фильтр для постепенной смены цвета

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+}
+
+.button__save-icon {
+  block-size: 21px;
```
**Надо исправить** 
 Для блоков с динамически изменяющимся контентом не следует создавать жесткое ограничение высоты. Если контента будет много он вывалится из блока, если мало, то блок будет почти пуст. 
 
 * Если блок содержит изображение на фоне или абсолютно спозиционированные элементы, то тогда лучше используйте свойство `min-height`.  https://doka.guide/css/height/#kak-eto-ponyat 
 * Если блок содержит текстовой контент не используется указание высоты совсем. 
 * Вместо фиксированной высоты хорошо кнопкам добавлять внутренние отступы `padding`

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  gap: 4px;
+}
+
+.button__text {
```
**Надо исправить** Опишите класс для базовой кнопки `button` и для ее контента `button__span`, указав все параметры по макету, остальные доп классы используйте только для уточнения, если есть различия. Дублирования следует удалить.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+}
+
+.card__like-button:hover {
+  transition: color 1s ease;
```
**Надо исправить** Перенесите параметр плавности в базовый класс кнопки, и скорректируйте, т.к. по заданию, плавность здевсь необходима только для тени с нужным временем.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  transition: color 1s ease;
+}
+
+.card__icon-button {
```
**Надо исправить** Добавьте плавность перехода для границы

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  padding: 8px 0;
+  position: relative;
+  overflow: hidden;
+  block-size: 37px;
```
**Надо исправить**
 Для блоков с динамически изменяющимся контентом не следует создавать жесткое ограничение высоты. Если контента будет много он вывалится из блока, если мало, то блок будет почти пуст.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+.card__icon-button:focus-visible {
+  border: 2px solid var(--accent-color);
+  padding-inline: 2px;
+  transition: border 0.3s ease;
```
**Надо исправить** Перенесите сво-во в базовый класс кнопки

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  outline: none;
+}
+
+.card__icon-button:focus-visible {
```
**Надо исправить** https://yadi.sk/d/ZdkTFL8NyRKuRg в фокусе высота равняется высоте кнопки

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
+  }
+}
+
+.sparks {
```
**Надо исправить** Перенесите описание искр в файл animations.css

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,390 @@
 .page {
```
**Надо исправить**
 https://yadi.sk/d/Bj28Ij8EeotTiw

Обратите внимание, что при сужении экрана образуется горизонтальный скролл. Это чаще всего происходит когда элементы не влезают по ширине, стоит проверить фиксированные ширины, размеры заголовков и перестройку сетки на брейкпоинтах, внести соотв. изменения. Помните также, что медиа запросы должны быть сгруппированы, не менее 150-200п разницей.

Размеры контента и логотипа можно скорректировать в медиазапросе 375

Savallera reviewed (changes requested):
## Здравствуйте! У Вас отличная работа, что отметила 


### Положительные моменты:

* Все комментарии в коде с пометкой **Отлично**.
* Нет ошибок валидации, а также отсутствуют ошибки в консоли
* Шрифты корректно подключены, нет лишних файлов.
* Для разметки выбраны верные семантические теги.
* Корректно работают цветовые темы
* Изображения не искажаются и выглядят как в брифе.
* Цветовая гамма, свойства шрифтов выдержаны по макету.


### Что требует доработки:

* Все комментарии в коде с пометкой **Надо исправить** 
* Проект требуется разместить на GitHub Pages и в README указать ссылку

### Что можно сделать лучше:

* Все комментарии в коде с пометкой **Можно лучше**
* Для всех функций  `var()`  в CSS нужно задать резервные значения,  на случай если доступ к кастомному свойству по имени будет невозможен. Подробнее https://doka.guide/css/custom-properties/
* [Как создать липкий футер с помощью flexbox](https://webformyself.com/kak-sozdat-lipkij-futer-s-pomoshhyu-flexbox/)



    Вносите правки без спешки, следите за тем, что изменилось после внесения той или иной правки в элементе. Убедитесь перед отправкой, что все правки внесены, основные требования соответствуют брифу и чек листу. Если есть трудности - обращайтесь к наставнику и старшему студенту, они обязательно Вам помогут.

    Комментарии **Надо исправить** обязательны для исправления.
    Комментарии **Можно лучше** носят рекомендательный характер, их выполнение остается на ваше усмотрение. 


**У Вас все получится! Удачного дня!**

AlinaLyubeznova committed:
Внесены правки согласно замечаниям ревьюера

AlinaLyubeznova committed:
Внесены правки согласно замечаниям ревьюера

Тесты выдали ошибку:
Element “p” not allowed as child of element “button” in this context.
Поэтому поменяла элементы <p> в кнопках обратно на <span>.

Savallera commented on index.html:
```diff
@@ -39,7 +45,421 @@
+            <button
+              class="button dialog_button"
+              type="submit">
+              <span class="dialog_text">ok</span>
```
**Надо исправить** Используйте базовый класс для спана внутри кнопки `button__text`, а дополнительный для коррекции

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,95 @@
+  transition: fill 0.1s linear;
+}
+
+.like-icon.is-liked .heart {
```
**Надо исправить** Удалите селектор `.heart` для того, чтобы эффект анимации работал на все сердечко, вместе с искрами

Savallera commented on styles/animations.css:
```diff
@@ -12,3 +12,95 @@
+
+.like-icon.is-liked .heart {
+  animation: heart-scale 0.3s 0.1s ease-in alternate;
+  transform-origin: center;
```
**Надо исправить** Данное сво-во опишите в базовом классе, для того, чтобы эффект корректировался сразу до появления класса `.like-icon {}`

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+  mix-blend-mode: difference;
+}
+
+.button:hover {
```
**Надо исправить** Опишите плавность перехода в базовом сво-ве кнопки `button`, и скорректируйте по заданию, в данном задании при наведении кнопка ничего не делает (наезжает прямоугольник на псевдоэлементе), переход необходим для тени в фокусе.

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+
+.card__like-button {
+  padding: 0px 35px;
+  max-block-size: 37px;
```
**Надо исправить** 
 Для блоков с динамически изменяющимся контентом не следует создавать жесткое ограничение высоты. Если контента будет много он вывалится из блока, если мало, то блок будет почти пуст. 
 
 * Если блок содержит изображение на фоне или абсолютно спозиционированные элементы, то тогда лучше используйте свойство `min-height`.  https://doka.guide/css/height/#kak-eto-ponyat 
 * Если блок содержит текстовой контент не используется указание высоты совсем. 
 * Вместо фиксированной высоты хорошо кнопкам добавлять внутренние отступы `padding`

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+
+.card__icon-button:hover,
+.button:hover {
+  cursor: pointer;
```
**Можно лучше** Опишите сво-во один раз в базовом классе кнопки `button` и `card__icon-button`, чтобы удалить перечисление классов здесь

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+
+.button:focus-visible {
+  box-shadow: 2px 2px 0px 0px var(--accent-color, #000);
+  transition: box-shadow 0.3s ease;
```
**Надо исправить** Данное сво-во переместите в базовый класс кнопки `button`

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+    gap: 49px;
+  }
+
+  .card__save-button {
```
**Надо исправить** 
   Дублирование свойства в медиазапросе, их следует удалять, это лишние строчки кода, которые ведут к нагромождению, описывайте только изменений в свойствах. [Подробнее](https://yoksel.github.io/bad-practices/styles-in-mediaquery/)

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+}
+
+.card__like-button {
+  padding: 0px 35px;
```
**Надо исправить** https://yadi.sk/d/Yen5QVzGwApepQ скорректируйте горизонтальные (`inline`) отступы, чтобы при смене текста ширина кнопки не менялась

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
 .page {
   min-inline-size: 320px;
```
**Надо исправить** https://yadi.sk/d/Bj28Ij8EeotTiw

Обратите внимание, что при сужении экрана образуется горизонтальный скролл. Это чаще всего происходит когда элементы не влезают по ширине, стоит проверить фиксированные ширины, размеры заголовков и перестройку сетки на брейкпоинтах, внести соотв. изменения. Помните также, что медиа запросы должны быть сгруппированы, не менее 150-200п разницей.

Размеры контента и логотипа можно скорректировать в медиазапросе 375

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+}
+
+.card__buttons {
+  padding: 25px 23px 20px 26px;
```
**Надо исправить** https://yadi.sk/d/2EpDSJA0ZXu53w / https://yadi.sk/d/s-n5lu40CLQ9ww требуется небольшая коректировка нижнего отступа по макету

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+
+.card__save-button {
+  padding: 12px 22px;
+  margin-block: 49px;
```
**Надо исправить** https://yadi.sk/d/xg7LDB2J5oWmvg / https://yadi.sk/d/NGO7pPIGLB30Sw отступ от последней карточки не по макету

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+  background-color: rgb(0 0 0 / 0.75);
+}
+
+.dialog {
```
**Надо исправить** 

1. https://yadi.sk/i/PWUHR_1fel0C8Q размеры модального окна отличаются от макета 1440

2. https://yadi.sk/d/SANTGJwQx6XxdA размеры модального окна и отступы отличаются от макета 375

Savallera commented on styles/style.css:
```diff
@@ -17,4 +17,372 @@
+    gap: 5px;
+  }
+
+  .button__save-icon {
```
**Надо исправить** https://yadi.sk/d/78qlYaWumGiK8Q размеры дискетки в кнопке "Сохранить на память" не по макету 375

Savallera reviewed (changes requested):
## Здравствуйте! Спасибо, что уделили внимание комментариям и внесли исправления. Ваша работа принимается на 2 итерации. 

В коде оставила Вам комментарии, с пометкой Надо исправить, чтобы результат был еще лучше. 

Спасибо вам за терпение и приложенные усилия. У Вас получился замечательный результат. Желаю дальнейших успехов!

---

**Полезности для Вас**

    - При организации внутренней геометрии блоков внутренние отступы задаются через padding https://doka.guide/css/padding/
    - Какие единицы измерения использовать для line-height? https://melikhova-summary.gitbook.io/veb-tekhnologii/html-sss/dekorativnye-elementy/kakie-edinicy-izmereniya-ispolzovat-dlya-line-height
    - 7 простых способов писать по-настоящему ужасный CSS https://proglib.io/p/terrible-css
